scripts/generatePlatformConfigurationProps.ts

```diff
@@ -0,0 +1,420 @@
+  defaultExplanation: string | null;
+  removedIn: string | null;
+  children?: Data[];
+  prefix: string | null;
+}
+
+type ConfigFileFormat = 'yaml' | 'env' | 'properties';
+type DisplayOption = 'kebab-case' | 'snake-upper-case';
+type ConfigFormat = {
+  value: ConfigFileFormat;
+  label: string;
+  displayOption: DisplayOption;
+};
+
+interface Definition {
+  url: string;
+  filename: string;
+  configFormats: ConfigFormat[];
+}
+
+const definition: Definition = {
+  url: 'http://localhost:8080/v2/public/configuration-properties',
+  filename: 'platformConfigurationProps.json',
+  configFormats: [
+    { value: 'yaml', label: 'YAML', displayOption: 'kebab-case' },
+    { value: 'env', label: '.env', displayOption: 'snake-upper-case' },
+    {
+      value: 'properties',
+      label: 'application.properties',
+      displayOption: 'kebab-case',
+    },
+  ],
+};
+
+downloadProps().then(() => {
+  generateDocsMdx();
+});
+
+function camelCaseToKebabCase(str: string): string {
+  return str.replace(/([A-Z])/g, (match) => `-${match[0].toLowerCase()}`);
+}
+
+function camelCaseToSnakeUpperCase(str: string): string {
+  return str
+    .replace(/([A-Z])/g, (match) => `_${match[0]}`)
+    .replace(/(\.|-)/g, '_')
+    .toUpperCase();
+}
+
+function getName(item: Data, displayOption: DisplayOption) {
+  if (!item.displayName) {
+    return getNameWithoutDisplayName(item, displayOption);
+  } else {
+    return item.displayName;
+  }
+}
+
+function getNameWithoutDisplayName(item: Data, displayOption: DisplayOption) {
+  let name: string;
+  if (displayOption === 'kebab-case') {
+    name = item.prefix === null ? item.name : camelCaseToKebabCase(item.name);
+  } else if (displayOption === 'snake-upper-case') {
+    name =
+      item.prefix === null ? item.name : camelCaseToSnakeUpperCase(item.name);
+  } else {
+    name = item.name;
+  }
+  return name;
+}
+
+function writeItems(
+  items: Data[],
+  stream: fs.WriteStream,
+  configFormatIndex: number,
+  level = 2
+) {
+  items.forEach((item) => {
+    const displayOption =
+      definition.configFormats[configFormatIndex].displayOption;
+    const name = getName(item, displayOption);
+
+    if (item.children) {
+      if (configFormatIndex === 0) {
+        stream.write(`${'#'.repeat(level)} ${name}`);
+      } else {
+        const slugger = createSlugger();
+        stream.write(
+          `<h${level} id="${slugger.slug(name)}">${name}</h${level}>`
+        );
+      }
+      stream.write('\n\n');
+    } else {
+      if (configFormatIndex === 0) {
+        stream.write(
+          `- ${'#'.repeat(5)} ${
+            item.removedIn ? `~~\`${name}\`~~` : `\`${name}\``
+          } {#${
+            displayOption === 'snake-upper-case'
+              ? getName(item, 'kebab-case')
+              : name
+          }}`
+        );
+      } else {
+        stream.write(`- <h5 id="${getName(item, 'kebab-case')}">`);
+
+        if (item.removedIn) stream.write('<del>');
```

Comment:
This kind of if statements are bad for debugging. Prettier would probably fix it. We always use curly brackets with ifs.